### PR TITLE
Split cats and cats-effect off into expensive-ci

### PR DESCRIPTION
**Pull request by valencik**
This PR creates two Steward jobs, one for "expensive" CI and one for "cheap" CI.
With "expensive" and "cheap" being differentiated by how big their build matrices are.

"cheap" CI runs every 20min from 3am to midnight
And "expensive" CI runs from midnight to 3am

From https://devtoolcafe.com/tools/cron:

`*/20 0-2  * * *`:  ~Every 20 minutes, between 00:00 and 02:59, every day~
`*/20 3-23 * * *`: ~Every 20 minutes, between 03:00 and 23:59, every day~

Update, new times:

`*/30 3-5 * * *`: Every 30 minutes, between 03:00 and 05:59, every day
`*/20 6-2 * * *`: Every 20 minutes, between 06:00 and 02:59, every day

We also limit Steward to only opening one PR per run

### CODE REVIEW

**Comment by valencik** [comment]
Oh, we probably want to set `updates.limit` if we're gonna bump the frequency up to 20min...

File: expensive-ci.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+- typelevel/cats-effect:series/3.x
+- typelevel/fs2
```
Comment (armanbilge) [inline expensive-ci.md:2]: Cats is worse than fs2 right now actually.

Comment (valencik) [reply expensive-ci.md:2]: Is this an endorsement for cats inclusion or fs2 exclusion? :P

File: .github/workflows/steward.yml [hunk 2]
```diff
@@ -2,16 +2,39 @@ name: Typelevel Steward
...
+    - cron: '*/20 6-2 * * *'
+    - cron: '*/30 3-5 * * *'
   workflow_dispatch:
 
 jobs:
-  steward:
+  steward-cheap:
     runs-on: ubuntu-latest
     name: Typelevel Steward
+    if: github.event.schedule != '*/30 3-5 * * *'
```
Comment (armanbilge) [inline .github/workflows/steward.yml:13]: This filter will prevent the `workflow_dispatch` trigger from working, we should check for that too.

Comment (valencik) [reply .github/workflows/steward.yml:13]: Sorry, I don't think I follow.
It's actually not clear to me me what the `workflow_dispatch` is 😬

Comment (armanbilge) [reply .github/workflows/steward.yml:13]: It's the thing that lets you manually trigger the workflow :)

Although, now that its running every 20 minutes it's not really important maybe ...

Comment (armanbilge) [reply .github/workflows/steward.yml:13]: Oh wait, it's a `!=`, not a `==`. Ignore me 😂

**Review by armanbilge — APPROVED** [review]
Let's give a try. Thanks for working on this!

**Comment by armanbilge** [comment]
Hmm.
```
 Invalid workflow file: .github/workflows/steward.yml#L1
invalid `cron` attribute "*/20 6-2 * * *"
```

**Comment by valencik** [comment]
So the dash syntax is part of the posix standard, but maybe I'm not allowed to loop around like in `6-2`
I'll experiment more tomorrow

**Comment by armanbilge** [comment]
Already fixed in https://github.com/typelevel/steward/pull/19 :)

**Pull request merged**